pylint/checkers/imports.py

```diff
@@ -914,8 +932,10 @@ def _check_reimport(
         basename: str | None = None,
         level: int | None = None,
     ) -> None:
-        """Check if the import is necessary (i.e. not already done)."""
-        if not self.linter.is_message_enabled("reimported"):
+        """Check if a module with the same name is already imported or aliased."""
+        if not self.linter.is_message_enabled(
+            "reimported"
+        ) and not self.linter.is_message_enabled("shadowed-import"):
```

Comment:
I see a few similar is_message_enabled` checks in the codebase, should these actually be replaced with `@only_required_for_messages(...`?

Reply:
I think ``only_required_for_messages`` only work for visitor function, so ``is_message_enabled`` is a suboptimization for non visitor function. Imo it's premature optimization to add them when implementing but also making the code worst to remove them so we should just leave it alone unless there's an actual problem with it.